Kudu.Core/Helpers/SiteTokenHelper.cs

```diff
@@ -0,0 +1,87 @@
+//------------------------------------------------------------------------------
+// <copyright file="SiteTokenHelper.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using Kudu.Core.Settings;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Kudu.Core.Helpers
+{
+    /// <summary>
+    /// Helper to issue and authenticate x-ms-site-token between antares role and site communication.
+    /// </summary>
+    public static class SiteTokenHelper
+    {
+        public const string SiteTokenHeader = "x-ms-site-token";
+
+        private static string _siteTokenAudience;
+        private static string _siteTokenIssuer;
+        private static TimeSpan? _siteTokenValidity;
+
+        // Setter for negative testing purposes
+        public static string SiteTokenAudience
+        {
+            get => _siteTokenAudience ?? "https://appservice.core.azurewebsites.net";
+            set => _siteTokenAudience = value;
+        }
+
+        public static string SiteTokenIssuer
+        {
+            get => _siteTokenIssuer ?? "https://appservice.core.azurewebsites.net";
+            set => _siteTokenIssuer = value;
+        }
+
+        public static TimeSpan? SiteTokenValidity
+        {
+            get => _siteTokenValidity ?? TimeSpan.FromHours(1);
+            set => _siteTokenValidity = value;
+        }
+
+        public static string IssueToken(TimeSpan? validity = null)
+        {
+            var utcNow = DateTime.UtcNow;
+            var expires = utcNow.Add(validity.GetValueOrDefault(SiteTokenValidity.Value));
+
+            return new JwtSecurityTokenHandler().CreateEncodedJwt(
+                issuer: SiteTokenIssuer,
+                audience: SiteTokenAudience,
+                subject: null,
+                notBefore: (expires > utcNow) ? utcNow : expires.AddHours(-1),
+                expires: expires,
+                issuedAt: (expires > utcNow) ? utcNow : expires.AddHours(-1),
+                signingCredentials: new SigningCredentials(new SymmetricSecurityKey(GetWebSiteAuthEncryptionKey()), SecurityAlgorithms.HmacSha256Signature));
+        }
+
+        // SiteTokenIssuingMode
+        // 0: (default) add both x-ms-site-restricted-token and x-ms-site-token
+        // 1: add x-ms-site-restricted-token only
+        // 2: add x-ms-site-token only
+        public static bool ShouldAddSiteRestrictedToken() => 2 != ScmHostingConfigurations.SiteTokenIssuingMode;
+        public static bool ShouldAddSiteToken() => 1 != ScmHostingConfigurations.SiteTokenIssuingMode;
+
+        private static byte[] GetWebSiteAuthEncryptionKey()
```

Comment:
Rather than copy this code from https://github.com/projectkudu/kudu/blob/98ad238b860f81a4cb4e3419c8785a58ba68e661/Kudu.Core/Helpers/SimpleWebTokenHelper.cs#L57, can we share it across both classes?

Reply:
We will get rid off the other class eventually.   So duplicate method for now.